examples/finite_prime_field.rs

```diff
@@ -0,0 +1,256 @@
+use noether::{
+    AssociativeAddition, AssociativeMultiplication, CommutativeAddition, CommutativeMultiplication,
+    DistributiveAddition, DistributiveMultiplication, FiniteField,
+};
+use num_traits::{Euclid, Inv, One, Zero};
+use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Rem, Sub, SubAssign};
+
+/// The trait hierarchy in Noether is useful for verifying the correct and idiomatic implementation
+/// of algebraic structures in rust.
+///
+/// We take here, as an example, the implementation of a finite field.
+
+#[derive(Clone, Copy, Debug)]
+pub struct FinitePrimeField<const L: usize, const D: usize> {
+    modulus: [u64; L],
+    _value: [u64; L],
+}
+
+impl<const L: usize, const D: usize> FinitePrimeField<L, D> {
+    const _ZERO: [u64; L] = Self::zero_array();
+
+    pub const fn new(modulus: [u64; L], value: [u64; L]) -> Self {
+        if D != 2 * L {
+            panic!("Double size D must be twice the size of the field L");
+        }
+        Self {
+            modulus,
+            _value: value,
+        }
+    }
+
+    /// Creates an array representing zero in the field.
+    ///
+    /// # Returns
+    ///
+    /// An array of L u64 elements, all set to 0
+    const fn zero_array() -> [u64; L] {
+        [0; L]
+    }
+}
+
+impl<const L: usize, const D: usize> Add for FinitePrimeField<L, D> {
+    type Output = Self;
+
+    /// Performs modular addition.
+    ///
+    /// This method adds two field elements and reduces the result modulo the field's modulus.
+    fn add(self, _other: Self) -> Self {
```

Comment:
So this operation adds two elements from the same field? But each is a FinitePrimeField object?

Reply:
Yes, it would if implemented, again, this is just a skeleton, see the other implementation for that detail.

Reply:
I get a little thrown by having an element be of type *Field. One could imagine having members/methods on the field itself vs elements. In another language I'd probably nest an Element class to have something like FiniteField<size=17>::Element(3), but again I don't have my head wrapped around the greater use case.

Reply:
Yeah, it's a bit circular. The finite prime field here is the element in a finite prime field for sure, but there isn't a reason you'd want an enumeration of every element in the field afaict, you'd just want to be able to do arithmetic with elements from the field in an idiomatic way.